File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
	restult := make([]int, 2)
	mapNums := make(map[int]int)
	for i, num := range nums {
		val, ok := mapNums[target-num]
		if ok {
			restult[0] = val
			restult[1] = i
			return restult
		} else {
			mapNums[num] = i
		}
	}
	return restult
}

File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val seen = mutableMapOf<Int,Int>()
        for(i in nums.indices){
            if(seen.contains(target - nums[i])){
                return intArrayOf(seen[target - nums[i]]!!,i)
            } else {
                seen[nums[i]] = i
            }
        }
        return intArrayOf()
    }
}